src/Sign.Cli/AzureAuthOptions.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE.txt file in the project root for more information.
+
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using Azure.Core;
+using Azure.Identity;
+using Sign.Core;
+
+namespace Sign.Cli
+{
+    internal sealed class AzureAuthOptions
+    {
+        internal Option<bool> ManagedIdentityOption { get; } = new(["-azm", "--azure-managed-identity"], getDefaultValue: () => false, Resources.ManagedIdentityOptionDescription);
+        internal Option<string?> TenantIdOption { get; } = new(["-azt", "--azure-tenant-id"], Resources.TenantIdOptionDescription);
+        internal Option<string?> ClientIdOption { get; } = new(["-azi", "--azure-client-id"], Resources.ClientIdOptionDescription);
+        internal Option<string?> ClientSecretOption { get; } = new(["-azs", "--azure-client-secret"], Resources.ClientSecretOptionDescription);
```

Comment:
Do we really want to top-level this option? There are many better options than client secret these days. IMHO we shouldn't have that option and force the user to, say, use Az CLI with client secret (they do this outside of Sign CLI) then depend on `DefaultAzureCredential` picking up that state.

Reply:
I think part of my concern here is that the client secret flow is the only one with explicit support in the options, but other great options exist in the Azure.Identity library. So we're sort of picking a winner which IMO is the least ideal from a security perspective.

Reply:
This is a duplicate of https://github.com/dotnet/sign/pull/716#discussion_r1644941441.